src/Streamly/Coreutils/FileTest.hs

```diff
@@ -306,47 +334,97 @@ isSticky = undefined
 --
 -- Like coreutil @test -r file@
 --
--- /Unimplemented/
-isReadable :: FileTest
-isReadable = undefined
-
--- | True if file exists and is writable.  True indicates
--- only that the write flag is on.  The file is not writable on a read-only
--- file system even if this test indicates true.
---
--- Like coreutil @test -w file@
---
--- /Unimplemented/
-isWritable :: FileTest
-isWritable = undefined
-
--- | True if file exists and is executable.  True indicates
--- only that the execute flag is on.  If file is a directory, true
+-- /Pre-release/
+
+isAccessible :: Tuple3 FileMode FileMode FileMode -> FilePath -> IO FileTest
+isAccessible (Tuple3 onr grp oth) path =  do
+    ownerIDMatch <- isOwnerEUID
+    groupIDMatch <- isGroupEGID
+    let ownerPerm = FileTest $ hasMode onr
+        groupPerm = FileTest $ hasMode grp
+        otherPerm = FileTest $ hasMode oth
+    ownerHasPermission <- test path ownerPerm
+    isOwner <- test path ownerIDMatch
```

Comment:
Calling test multiple times will add the overhead of making the `stat` system call multiple times. We can write the predicate in a way so that it does not have to use `test` multiple times. It may possibly require the predicate to be of type `a -> m Bool` rather the `a -> Bool`. I will have to look into it.